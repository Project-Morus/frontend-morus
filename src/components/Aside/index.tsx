import { ContainerSC, ContentSC, FooterSC } from "./styles";
import { IAsideProps } from "./types";
import LogoWhiteSvg from "../../assets/svg/LogoWhite";

const Aside = ({ position = 'right', ...props }: IAsideProps) => {
  const roleSVG = 'logoMorusWhite'

  return (
    <ContainerSC $position={position} role={props.role}>
      <ContentSC $position={position}>
        <LogoWhiteSvg role={roleSVG} />
      </ContentSC>

      {props.children}

      <FooterSC $position={position}>Todos os direitos reservados Morus Company 2023 | Vers√£o 1.0.0</FooterSC>
    </ContainerSC>
  );
}

export default Aside;