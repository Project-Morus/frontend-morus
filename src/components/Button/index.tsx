import { ReactNode } from "react";
import { IButtonProps, VariantsTypes } from "./types";
import ButtonPrimary from "./variants/Primary";
import ButtonSecondary from "./variants/Secondary";
import ButtonRed from "./variants/Red";
import ButtonGreen from "./variants/Green";

const Button = ({ variant = "primary", ...props }: IButtonProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <ButtonPrimary {...props} />,
    secondary: <ButtonSecondary {...props} />,
    red: <ButtonRed {...props} />,
    green: <ButtonGreen {...props} />,
  };

  return variantType[variant];
};

export default Button;
