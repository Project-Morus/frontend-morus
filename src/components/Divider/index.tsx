import { ReactNode } from "react";
import { IDividerProps, VariantsTypes } from "./types";
import DividerPrimary from "./variants/Primary";
import DividerSecondary from "./variants/Secondary";

const Divider = ({ variant = "primary", ...props }: IDividerProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <DividerPrimary {...props} />,
    secondary: <DividerSecondary {...props} />,
  };

  return variantType[variant];
};

export default Divider;
