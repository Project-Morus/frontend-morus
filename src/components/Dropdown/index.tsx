import { useState } from "react";
import { DropdownContainer, DropdownButton, DropdownMenu, DropdownItem, LabelContainer } from "./styles";
import { DropdownProps } from "./types";
import Icon from "../DataDisplay/Icon";
import { useTheme } from "styled-components";

function Dropdown({ label, options }: DropdownProps) {
  const [isOpen, setIsOpen] = useState(false);
  const theme = useTheme();

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  return (
    <DropdownContainer>
      <DropdownButton onClick={toggleDropdown}>
        <LabelContainer>
          {label}
          {isOpen ? (
            <Icon name="ph-caret-up" size="16px" stroke="bold" color={theme.colors.primary[500]} />
          ) : (
            <Icon name="ph-caret-down" size="16px" color={theme.colors.primary[500]} />
          )}
        </LabelContainer>
      </DropdownButton>
      {isOpen && (
        <DropdownMenu>
          {options.map((item: string, index: number) => (
            <DropdownItem onClick={toggleDropdown} key={index}>
              {item}
            </DropdownItem>
          ))}
        </DropdownMenu>
      )}
    </DropdownContainer>
  );
}

export default Dropdown;
