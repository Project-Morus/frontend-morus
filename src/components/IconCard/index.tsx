import Icon from "../Icon";
import { CardSC } from "./styles";
import { IIconCardProps } from "./types";

const IconCard = ({ icon, stroke, variant = 'white', ...props }: IIconCardProps) => {

  return (
    <CardSC $variant={variant}
      role={props.role}
      className={props.className}
      data-testid={props?.['data-testid']}
    >
      <Icon
        name={icon}
        size="47"
        stroke={stroke}
      />
    </CardSC>
  );
}

export default IconCard;