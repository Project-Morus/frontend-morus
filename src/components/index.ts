
// Buttons
export { default as Button } from './Buttons/Button'
export { default as RadioButtonGroup } from './Buttons/RadioButtonGroup'
export { default as RadioButton } from './Buttons/RadioButton'

// DataDisplay
export { default as Icon } from './DataDisplay/Icon'
export { default as IconCard } from './DataDisplay/IconCard'
export { default as SequencialStepper } from './DataDisplay/SequencialStepper'
export { default as Table } from './DataDisplay/Table'

export { default as Archive } from './DataDisplay/Archive'

// Feedback
export { default as Aside } from './Feedback/Aside'
export { default as Divider } from './Feedback/Divider'
export { default as Modal } from './Feedback/Modal'


// Inputs
export { default as Input } from './Inputs/Input'

// Navigations
export { default as Menu } from './Navigation/Menu';
