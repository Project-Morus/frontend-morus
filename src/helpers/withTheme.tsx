import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

