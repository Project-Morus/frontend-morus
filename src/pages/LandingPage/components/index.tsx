export * from './InformationSection';
export * from './HeroSection';
export * from './CommonAreaSection';
export * from './CustomersSection';
export * from './FooterSection';