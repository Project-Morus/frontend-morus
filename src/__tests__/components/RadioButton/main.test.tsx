import { render, screen } from "@testing-library/react";
import { RadioButton } from "../../../components";
import { withTheme } from "../../../helpers/withTheme";

const RadioButtonWithTheme = withTheme(RadioButton);

const idName = "idname";
const role = "radiogroup";

describe("RadioButton Component", () => {
  test("Verify if component is rendering", () => {
    render(<RadioButtonWithTheme label="Name" id={idName} role={role} />);


    expect(screen.getByRole(role)).toBeVisible();
  });

  test("Verify if label is rendering correctly", () => {
    render(<RadioButtonWithTheme id={idName} label="Name" role={role} />);
    const labelElement = screen.getByText("Name");

    expect(labelElement).toBeVisible();
  });
  test("Verify if RadioButton disabled is rendering correctly", () => {
    render(<RadioButtonWithTheme id={idName} disabled label="Name" role={role} />);
    const radioButtonElement = screen.getByRole("radiogroup");
    expect(radioButtonElement).toBeDisabled();
  });

  test("Snapshot RadioButton", () => {
    const { asFragment } = render(<RadioButtonWithTheme id={idName} label="Name" role={role} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
