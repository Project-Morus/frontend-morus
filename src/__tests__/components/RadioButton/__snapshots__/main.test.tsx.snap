// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component Snapshot RadioButton 1`] = `
<DocumentFragment>
  .c1 {
  width: 20px;
  height: 20px;
  appearance: none;
  border-radius: 50%;
  border: 2px solid #023B74;
  cursor: pointer;
  transition: border-color 0.2s;
}

.c1:checked {
  border-width: 5px;
}

.c1:not(:checked):hover {
  border-color: #0353A4;
}

.c1:disabled {
  opacity: 0.5;
  cursor: default;
}

.c0 {
  width: 100%;
  display: flex;
  gap: 0.5rem;
  border: 1px solid #BABABA;
  border-radius: 5px;
  background-color: #FFF;
  padding: 1rem;
}

.c2 {
  color: #4E4E4E;
}

<div
    class="c0"
  >
    <input
      class="c1"
      id="idname"
      role="radiogroup"
      type="radio"
    />
    <label
      class="c2"
      for="idname"
    >
      Name
    </label>
  </div>
</DocumentFragment>
`;
